hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/HoodieFileGroupReaderBasedParquetFileFormat.scala

```diff
@@ -300,12 +303,13 @@ class HoodieFileGroupReaderBasedParquetFileFormat(tableState: HoodieTableState,
     val baseFileReader = super.buildReaderWithPartitionValues(sparkSession, dataSchema, partitionSchema, requiredSchema,
       filters ++ requiredFilters, options, new Configuration(hadoopConf))
 
-    //file reader for reading a hudi base file that needs to be merged with log files
+    // File reader for reading a Hoodie base file that needs to be merged with log files
     val preMergeBaseFileReader = if (isMOR) {
       // Add support for reading files using inline file system.
-      super.buildReaderWithPartitionValues(sparkSession, dataSchema, partitionSchema, requiredSchemaWithMandatory,
-        if (shouldUseRecordPosition) requiredFilters else recordKeyRelatedFilters ++ requiredFilters,
-        options, new Configuration(hadoopConf))
+      val appliedRequiredSchema = sparkAdapter.appendRowIndexColumnForParquetFileReader(requiredSchemaWithMandatory, shouldUseRecordPosition)
```

Comment:
Could you add a test to validate the logic?

Reply:
Ok.

Reply:
maybe you can use `HoodieSparkUtils.gteqSpark3_5`